d3a_interface/sim_results/kpi.py

```diff
@@ -145,11 +145,50 @@ def update_area_kpi(self, area_dict, core_stats):
         self._accumulate_energy_trace(core_stats)
 
 
+class SavingKPI:
+    def __init__(self):
+        self.producer_list = list()
+        self.consumer_list = list()
+        self.ess_list = list()
+        self.house_list = list()
+        self.fit_rev = 0.
+        self.utility_bill = 0.
+        self.d3a_rev = 0.
+        self.saving_abs = 0.
+        self.saving_per = 0.
+
+    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
+        for child in area_dict['children']:
+            if is_producer_node_type(child):
+                if_not_in_list_append(self.producer_list, child['uuid'])
+            elif is_load_node_type(child):
+                if_not_in_list_append(self.consumer_list, child['uuid'])
+            elif is_prosumer_node_type(child):
+                if_not_in_list_append(self.ess_list, child['uuid'])
+
+        fir_excl_gf_alp = \
+            core_stats.get(area_dict['uuid'], {}).get('feed_in_tariff', 0.) - gf_alp
+        mmr_incl_gf_alp = \
```

Comment:
Name of variables should either be verbose or at least an inline comment needed

Reply:
I agree with Hassan about this :) Also, it would be good if you could use parentheses over backslashes in order to follow PEP-8 recommendations.

Reply:
Correct me if I'm wrong, isn't that proposal is rejected: https://www.python.org/dev/peps/pep-3125/

Reply:
The PEP you linked was to completely eliminate `\` as a marker for line continuation from python. It's been rejected, so the `\` still works, but python guidelines recommend avoiding it and use parentheses instead:
> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

Some exceptions are `assert` statements.